lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -349,6 +357,13 @@ static ValidationInfo validateControlBlock(
     }
   }
 
+  // Last resort check in cases where the format is broken enough that
+  // we didn't read the REVISION block, report such a case as incompatible.
+  if (requiresRevisionMatch &&
+      !revisionSeen &&
+      result.status == Status::Valid)
+    result.status = Status::RevisionIncompatible;
```

Comment:
Should it just be `Status::Malformed` in that case? Ie. the same as the missing version case?

Reply:
It could be, but I like having a different code for it. I would expect to see `Malformed` if the header is missing/truncated, while `RevisionIncompatible` remains a case where the format is mostly fine but it was definitively written by a different compiler.